cogs/mediaconverter.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import re
+from typing import Any, Dict
+
+import requests
+from disnake import Message
+from disnake.ext.commands import Cog
+
+from main import ManChanBot
+
+from .commandbase import CommandBase
+
+
+class Media_Converter(CommandBase):
+    @Cog.listener()
+    async def on_message(self, message: Message):
+        if message.author.bot:
+            return
+
+        link = self.extract_link(message.content)
+        if link:
+            if "tiktok.com" in link:
+                embedded_video = self.embed_tiktok(link)
+                await message.edit(
+                    suppress_embeds=True
+                )  # Removes previous TikTok embed from context message
+                await message.channel.send(f"[TikTok Link]({embedded_video})")
+            elif "twitter.com" in link or "x.com" in link:
+                converted_link = self.convert_twitter_link(link)
+                await message.channel.send(
+                    f"[Converted Twitter Link]({converted_link})"
+                )
+
+    @classmethod
+    def extract_link(cls, text: str):
```

Comment:
Why not have this return two variables?

`return "twitter", twitter_match.group(1)`
`return "tiktok", twitter_match.group(0)`

Then, in line 20. Instead of doing searches in a string, you can just compare if the first var is twitter or tiktok.

Reply:
I believe if you would then have to return None, None